mne/viz/backends/_pyvista.py

```diff
@@ -168,7 +168,6 @@ def __init__(self, fig=None, size=(600, 600), bgcolor='black',
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", category=FutureWarning)
             if MNE_3D_BACKEND_TESTING:
-                self.figure.plotter_class = Plotter
```

Comment:
Perhaps what you need to do when switching to the other plotting class is ensure that, during closure, events get processed before actually closing the window. For example back in the day for vispy I had to call `app.process_events()` [a couple of times](https://github.com/vispy/vispy/blob/master/vispy/app/backends/_qt.py#L770-L778), and this made things more stable / fixed segfaults.

But to test these sorts of things you really need to iterate locally on a machine with the OS rather than wait for CI 30-minute round trips.

BTW you can also speed up this iterative process on the CIs when you think you have something working by running just `pytest ... mne/viz` instead of `pytest ... mne`. Might be worth switching this to that now to ensure that the segfaults still happen with the reduced tests.